orchestrator/cli/domain_gen_helpers/product_block_helpers.py

```diff
@@ -0,0 +1,245 @@
+
+    Args:
+        - product_block_relations: Dict with product blocks by resource type
+            - key: product block name.
+            - value: Set of product block names to relate to.
+
+    Returns: .
+    """
+
+    def create_subscription_instance_relations(depends_block_name: str, block_names: Set[str]) -> List[str]:
+        depends_block_id_sql = get_product_block_id(depends_block_name)
+
+        def map_subscription_instances(block_name: str) -> List[Dict[str, Union[str, Query]]]:
+            in_use_by_id_sql = get_product_block_id(block_name)
+            subscription_ids: list[SubscriptionInstanceTable] = (
+                SubscriptionInstanceTable.query.where(SubscriptionInstanceTable.product_block_id.in_(in_use_by_id_sql))
+                .with_entities(SubscriptionInstanceTable.subscription_id)
+                .all()
+            )
+            if not subscription_ids:
+                return []
+            return [
+                {"subscription_id": subscription_instance.subscription_id, "product_block_id": in_use_by_id_sql}
+                for subscription_instance in subscription_ids
+            ]
+
+        def map_subscription_instance_relations(block_name: str) -> List[Dict[str, Union[str, Query]]]:
+            block_id_sql = get_product_block_id(block_name)
+            subscription_instance_ids: list[SubscriptionInstanceTable] = (
+                SubscriptionInstanceTable.query.where(SubscriptionInstanceTable.product_block_id.in_(block_id_sql))
+                .with_entities(SubscriptionInstanceTable.subscription_instance_id)
+                .all()
+            )
+            if not subscription_instance_ids:
+                return []
+            return [
+                {"in_use_by_id": instance_id.subscription_instance_id, "depends_on_id": depends_block_id_sql}
+                for instance_id in subscription_instance_ids
+            ]
+
+        subscription_instance_dicts = list(
+            flatten([map_subscription_instances(block_name) for block_name in block_names])
+        )
+        subscription_relation_dicts = list(
+            flatten([map_subscription_instance_relations(block_name) for block_name in block_names])
+        )
+
+        subscription_instances = ""
+        if subscription_relation_dicts:
+            subscription_instances = sql_compile(Insert(SubscriptionInstanceTable).values(subscription_instance_dicts))
+
+        subscription_block_relations = ""
+        if subscription_relation_dicts:
+            subscription_block_relations = sql_compile(
+                Insert(SubscriptionInstanceTable).values(subscription_instance_dicts)
+            )
+        return [subscription_instances, subscription_block_relations]
+
+    rt_relations = flatten(create_subscription_instance_relations(*item) for item in product_block_relations.items())
+    return [rt_relation for rt_relation in rt_relations if rt_relation]
```

Comment:
Similar to my comment in product_helpers, this could be condensed a bit by using `yield`-ing dicts/sql strings instead of wrapping them in lists, and then `list(chain.from_iterable(closure(arg) for arg in args))` instead of  `list(flatten([closure(arg) for arg in args]))` to join the yielded results

Optional though, depending on how much time you have left on this ticket